src/helpers/propPathOr.js

```diff
@@ -1,31 +1,45 @@
 /** @license ISC License (c) copyright 2017 original and current authors */
 /** @author Henrique Limas */
+/** @author Ian Hofmann-Hicks */
 
 const curry = require('../core/curry')
 const isArray = require('../core/isArray')
+const isEmpty = require('../core/isEmpty')
 const isInteger = require('../core/isInteger')
-const isNil= require('../core/isNil')
+const isDefined = require('../core/isDefined')
+const isNil = require('../core/isNil')
 const isString = require('../core/isString')
 
 // propPathOr : a -> [ String | Integer ] -> b -> c
 function propPathOr(def, keys, target) {
   if(!isArray(keys)) {
-    throw new TypeError('propPathOr: Array of strings or integers required for second argument')
+    throw new TypeError(
+      'propPathOr: Array of Non-empty Strings or Integers required for second argument'
+    )
   }
 
-  const value = keys.reduce((target, key) => {
-    if (isNil(target)) {
-      return target
-    }
+  if(isNil(target)) {
+    return def
+  }
+
+  let value = target
+  for(let i = 0; i < keys.length; i++) {
+    const key = keys[i]
 
-    if(!(isString(key) || isInteger(key))) {
-      throw new TypeError('propPathOr: Array of strings or integers required for second argument')
+    if(!((isString(key) && !isEmpty(key)) || isInteger(key))) {
```

Comment:
I'm not sure how much value this will add in early-exit cases (we discussed this in the `propPathEq` PR). But I guess some errors are better than no errors.

Reply:
Yep. And in the case of `propPathEq` it made sense. But as a whole, when all functions have been taken into account, it makes sense to add them. That is why I took a second to look at all of these functions as a whole.